unittest/unit/update-bcd.js

```diff
@@ -14,13 +14,728 @@
+          ['83', {result: null, prefix: ''}],
+          ['84', {result: null, prefix: ''}],
+          ['85', {result: null, prefix: ''}]
+        ])]])]
+      ]));
+
+      expect(console.warn.calledWith('Ignoring unknown browser/version: Mozilla/5.0 (Windows NT 6.3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 YaBrowser/17.6.1.749 Yowser/2.5 Safari/537.36')).to.be.true;
+    });
+
+    afterEach(() => {
+      console.warn.restore();
+    });
+  });
+
+  describe('inferSupportStatements', () => {
+    const expectedResults = {
+      'api.AbortController': [
+        {version_added: '≤83'}
+      ],
+      'api.AbortController.abort': [
+        {version_added: '≤84'}
+      ],
+      'api.AbortController.AbortController': [
+        {version_added: '85'}
+      ],
+      'api.AudioContext': [
+        {version_added: '85'}
+      ],
+      'api.AudioContext.close': [
+        {version_added: '85'}
+      ],
+      'api.DeprecatedInterface': [
+        {version_added: '≤83', version_removed: '85'}
+      ],
+      'api.ExperimentalInterface': [
+        {version_added: '≤83'}
+      ],
+      'api.NewInterfaceNotInBCD': [
+        {version_added: '85'}
+      ],
+      'api.PrefixedInterface1': [
+        {version_added: '≤83'}
+      ],
+      'api.PrefixedInterface2': [
+        {prefix: 'WebKit', version_added: '≤83'},
+        {version_added: '85'}
+      ],
+      'api.RemovedInterface': [
+        {version_added: '≤83', version_removed: '84'},
+        {version_added: '85'}
+      ],
+      'css.properties.font-family': [
+        {version_added: '84'}
+      ],
+      'css.properties.font-face': []
+    };
+
+    const supportMatrix = getSupportMatrix(bcd.browsers, reports);
+    for (const [path, browserMap] of supportMatrix.entries()) {
+      for (const [_, versionMap] of browserMap.entries()) {
+        it(path, () => {
+          assert.deepEqual(
+              inferSupportStatements(versionMap), expectedResults[path]
+          );
+        });
+      }
+    }
+
+    it('Invalid results', () => {
+      expect(() => {
+        const report = {
+          __version: '0.3.1',
+          results: {
+            'https://mdn-bcd-collector.appspot.com/tests/': [
+              {
+                name: 'api.AbortController',
+                info: {exposure: 'Window'},
+                result: 87
+              }
+            ]
+          },
+          userAgent: 'Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.61 Safari/537.36'
+        };
+        const versionMap = getSupportMatrix(bcd.browsers, [report])
+            .entries().next().value[1].entries().next().value[1];
+
+        inferSupportStatements(versionMap);
+      }).to.throw('result not true/false/null; got 87');
+    });
+  });
+
+  describe('update', () => {
+    const supportMatrix = getSupportMatrix(bcd.browsers, reports);
+    let bcdCopy;
+
+    beforeEach(() => {
+      bcdCopy = Object.assign({}, bcd);
```

Comment:
```suggestion
      bcdCopy = JSON.parse(JSON.stringify(bcd));
```

`Object.assign` results in a shallow copy, would would end up with interactions between tests. Using `JSON.parse(JSON.stringify(bcd))` is a quick if not pretty way to make a copy of a simple object.